flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/ScalarFunctionDefinition.java

```diff
@@ -53,6 +56,11 @@ public FunctionKind getKind() {
 		return FunctionKind.SCALAR;
 	}
 
+	@Override
+	public TypeInference getTypeInference(DataTypeFactory factory) {
```

Comment:
nit: 
1. The whole class should be `@Internal`
2. Could we add a comment on top of this class that this is a marker interface to use the old type stack for the function?